docs/source/build-with-bentoml/runtime-environment.rst

```diff
@@ -128,15 +128,39 @@ To configure PyPI indexes and other pip options:
             "torchaudio"
         )
 
-    # Multiple pip options
+    # Configuring multiple PyPI options, including a private repository
     image = bentoml.images.PythonImage(python_version='3.11') \
         .python_packages(
-            "--index-url https://pypi.org/simple",
-            "--extra-index-url https://my.private.pypi/simple",
-            "--trusted-host my.private.pypi",
+            "--index-url https://pypi.org/simple", # Default PyPI index
```

Comment:
@frostming I don't think this works with lock_packages=True

or uv don't support this use case yet.

Reply:
I have to do this https://github.com/bentoml/BentoVLLM/blob/c32bea67e2f15a960b9f45f193b65cc4292245ee/deepseek-r1-distill-llama3.1-8b-tool-calling/service.py#L33 (otherwise building on bentocloud would fail)

Reply:
Well yes. Why not use env var `UV_INDEX_URL`

Reply:
hmm because in case user want to specify `--extra-index-url` or `--find-links`, it can't seem to be resolved to a requirements.txt

fwiw it is also find to separate it out to a separate run step, to cache on image build.

Reply:
I mean you can use:

```python
python PythonImage().run("export UV_INDEX_URL=...").requirements_file("requirements.txt")
```

The commands are run in sequential in the same script so `export` works here, a small trick.

Reply:
lock won't work with `--index-url` in requirements.txt last time I tested with this.

Reply:
so it neither reads or writes that option in requirements.txt? well it's embarrassing

Reply:
or is this `--find-links`? iirc one of these options doesn't work with `lock` on bentocloud (the image builder failed at env-layer)

Reply:
Oh, the image builder doesn't lock but consumes the result of `lock` which is performed at `bentoml build`

The problem is it doesn't write that option into the result so it is missing when building image

Reply:
@frostming @aarnphm Just to follow up on this. Any change needed for the doc? Like using an env var (`UV_INDEX_URL`) during image building to ensure the index URL is properly set at install time:

```python
image = bentoml.images.PythonImage(python_version='3.11') \
          .run("export UV_INDEX_URL=https://my.private.pypi/simple") \
          .requirements_file("requirements.txt")
```

One thing I am thinking is that it may confuse users. Most people should just expect that specifying `--index-url` in `.python_packages()` would just work

Reply:
@Sherlock113 you can keep it as-is now. The latest bentoml version supports global options in `requirements.txt`. now